### ISSUE TITLE
There should be a delay before applying the autofocus

### ISSUE THREAD

**Issue by pgn-vole** [davidtheclark/focus-trap#46]
The autofocus of should be delayed to ensure that the focused item don't respond to the same keystroke that have activated the focus trap.

For instance
 1 - Press Enter on a button to show a modal
 2 - The modal get displayed with its close button focused
 3 - Releasing the Enter key closes the modal because the confirm button was focused

The issue there is that the same same event is caught by the OK button of the modal because it was focused within the same event cycle.
A solution would be to delay the autofocus to make sure it is not part of the same cycle.

You can find an example of this issue here:
https://codepen.io/anon/pen/MVZrxW

**Comment by davidtheclark** [comment]
@pgn-vole I'm not seeing the problem you describe. It's possible that the demo is buggy, or I just don't quite understand what I'm supposed to be looking for.

I do know that if you hold the Enter key, the browser is going to fire multiple events (just like you can hold a letter key to get a long string of that one letter). I don't think there's any way around that, though, because that's standard behavior.

**Comment by pgn-vole** [comment]
@davidtheclark sorry the example was not really good. I updated it with some explanation and also a way to fix the problem.

The issue I'm showing there is not is not related to the browser dispatching multiple events. It is that the same event is caught both by the open and close buttons and because the focus was given to the close button during the same event cycle.
A solution would be for focus trap to delay the autofocus.

**Comment by davidtheclark** [comment]
@pgn-vole I still don't understand the demo, and I'm not sure it changed. Is there a new URL?

**Comment by davidtheclark** [comment]
@pgn-vole Thanks for clearing that up.

Looks like that behavior is not reproducible in the demos because the demos use `click` handlers on the buttons, whereas your example uses `mousedown`. If you need a temporary workaround and you are using` <button>`s, you might try a `click` handler.

I think I'd be ok with applying a `setTimeout(() => {}, 0)` before the element is focused. I'd be open to a PR that adds a new demo to illustrate the problem, fixes the problem, and shows that all the existing demos still work the same as before.

**Comment by pgn-vole** [comment]
Looks like a plan. Will work on that.

**Comment by davidtheclark** [comment]
Closed by #51.
